### PR TITLE
Automatically install Dependencies and fix lint errors

### PR DESCRIPTION
**Pull request by LinusBorg**
## What this PR does

* runs `npm install`for you after you answered all the questions
* run `npm run lint -- --fix` to adjust the boilerplate code to the selected eslint config

This allows us to get rid of the ugly template strings we currently use to cater towards airbnb style, like this:
```handlebars
{{#if_eq lintConfig "airbnb"}},{{/if_eq}}
```

These make the template source files barely readable in places. Getting rid of them increases maintainability

## TODO

- [x] run `npm install` after the project directory has been set up.
- [x] run `npm run lint -- --fix` if airbnb preset was selected for eslint
- [x] Make auto-install optional through a question in the init questionnaire
- [x] If user chose `standard` or `airbnb` style and chose not to auto-install dependencies, we can't run lint-fix for him. In that case, print a not to run it themselves in the final console message.
- [x] Improve console messages with chalk
- [x] Let user choose between npm and yarn 

## How can you help?

Take this for a spin. I surely have made a few minor mistakes, like typos in the new console messages, and maybe you find a big one as well.

In particular, **testing it on windows** would be very helpful. 

If you find anything, just send a PR against this branch.

### CODE REVIEW

**Comment by sudo-suhas** [comment]
Hey @LinusBorg, I'd like to help you with testing this on windows. I'll try it out tomorrow. One suggestion I have is that it should be configurable to install using yarn. npm5 doesn't work very well on windows.

**Comment by LinusBorg** [comment]
Very good point, thanks. We could change the last question to be a choice:

> Install dependencies for you now (recommended)?
> o Ys, use npm
> o yes, use Yarn
> o no, I'll do that myself.

**Comment by sudo-suhas** [comment]
The install step did not work correctly:

```
λ vue init webpack#auto-install-deps vue-auto-install-deps

? Project name vue-auto-install-deps
? Project description A Vue.js project
? Author Suhas Karanth <sudo.suhas@gmail.com>
? Vue build runtime
? Install vue-router? Yes
? Use ESLint to lint your code? Yes
? Pick an ESLint preset Airbnb
? Set up unit tests Yes
? Pick a test runner jest
? Setup e2e tests with Nightwatch? Yes
? Should we run `npm install` for you after the project has been created? (recommended) yarn

   vue-cli · Generated "vue-auto-install-deps".


# Installing project dependencies ...
# ========================


events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: spawn yarn ENOENT
    at _errnoException (util.js:1024:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

I got the same error with npm as well.

**Comment by LinusBorg** [comment]
Hm, and am bus able stack trace. Dammit :D

**Comment by LinusBorg** [comment]
Hm, I see you created the project in place instead of defining a sub directory name.  I haven't tested that one tbh. 

Let's hope it's related to this and not windows.

**Comment by sudo-suhas** [comment]
I tried the other way to init:
```
λ mkdir vue-auto-install-deps

λ cd vue-auto-install-deps\

λ vue init webpack#auto-install-deps

? Generate project in current directory? Yes
? Project name vue-auto-install-deps
? Project description A Vue.js project
? Author Suhas Karanth <sudo.suhas@gmail.com>
? Vue build runtime
? Install vue-router? Yes
? Use ESLint to lint your code? Yes
? Pick an ESLint preset Airbnb
? Set up unit tests Yes
? Pick a test runner jest
? Setup e2e tests with Nightwatch? Yes
? Should we run `npm install` for you after the project has been created? (recommended) yarn

   vue-cli · Generated "vue-auto-install-deps".


# Installing project dependencies ...
# ========================


events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: spawn yarn ENOENT
    at _errnoException (util.js:1024:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**Comment by LinusBorg** [comment]
:(

**Comment by sudo-suhas** [comment]
Give me some hints. Maybe I can help you fix it.

**Comment by LinusBorg** [comment]
I can't give you much. It seems the `spawn()` of the child process in `/utils/index.js` fails, but I'm not very familiar with that API. Maybe a problem with the cwd? Maybe we have to specify another type of `shell` on windows systems because now it uses one where yarn & npm are not in the path?

**Comment by sudo-suhas** [comment]
Okay, I'll look into it.

File: utils/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * Options are the same as node's child_process.spawn
+ * @param {string} cmd 
+ * @param {array<string>} args 
+ * @param {object} options
+ */
+function runCommand(cmd, args, options) {
+  return new Promise((resolve, reject) => {
+    const spwan = spawn(cmd, args, Object.assign({
+      cwd: process.cwd(),
+      stdio: 'inherit',
```
Comment (LinusBorg) [inline utils/index.js]: Reading through a few old node issues like https://github.com/nodejs/node-v0.x-archive/issues/2318, i think we might just have to add `shell: true` here to make it run properly on windows.

Will test it tonight.

Comment (sudo-suhas) [reply utils/index.js]: I can confirm that this change fixes it.

Comment (LinusBorg) [reply utils/index.js]: Awesome!

**Comment by sudo-suhas** [comment]
btw.. Instead of manually wrapping `spawn` in a promise, why not use [`execa`](https://github.com/sindresorhus/execa)? It would handle a lot of edge cases as well.

**Comment by sudo-suhas** [comment]
Are you planning to drop support for node 4?

**Comment by LinusBorg** [comment]
> btw.. Instead of manually wrapping `spawn` in a promise, why not use [`execa`](https://github.com/sindresorhus/execa)? It would handle a lot of edge cases as well.

Because that would mean that we have to 
* first run an npm install in the temp dirctory where the template was downloaded to, to install `execa` (otherwise we install in the cwd which probably is undesirable)
* but I don't know if we can reliably know this direcory in the first place.
* say we do, then we can require  `exceca` from this directory 
* and finally we can run npm again in the created proejct folder to install the project dependencies.

A lot of hoops to jump through for convenience - I would only ttempt this if it really was necessary to handle edge cases without excessive boilerplate code.

> Are you planning to drop support for node 4?

I would like to, and vue-cli's latest version doesn't support it anymore anyways.

**Comment by sudo-suhas** [comment]
@LinusBorg The functionality that you have added to the template, it could be used by pretty much all templates right. So do you think it would make sense to implement it in the `vue-cli` instead?

**Comment by LinusBorg** [comment]
Not all need them, but there could be a flag to enable/disable it in meta.json - so yeah, that could make sense...

**Comment by LinusBorg** [comment]
I would still like to add it to the template first, and then integrate it in vue-cli.

Release cycles for he cli are longer, and I can always deactivate it in this tempate quickly once the feature has been integrated into the template.

this way, older cli versions can benefit from it as well...

**Pull request merged**